pkg/rpaas/client/types/types.go

```diff
@@ -190,6 +190,22 @@ type Bind struct {
 	Upstreams []string `json:"upstreams,omitempty"`
 }
 
+type UpstreamOptions struct {
+	PrimaryBind          string                        `json:"app"`
+	CanaryBinds          []string                      `json:"canary,omitempty"`
+	TrafficShapingPolicy v1alpha1.TrafficShapingPolicy `json:"trafficShapingPolicy,omitempty"`
+	LoadBalance          v1alpha1.LoadBalanceAlgorithm `json:"loadBalance,omitempty"`
+}
+
+type TrafficShapingPolicy struct {
```

Comment:
My LSP says that this struct here is not being used/referenced anywhere, so we might take it out?
Or maybe you meant to reference on line 196?